### PR TITLE
Change navigation and button labels

### PR DESCRIPTION
**Pull request by Mandily**
This is a precursor to #1120 

Some labels were inconsistent and some were confusing.
Based on a recent card sort and tree test on the navigation, I was able to figure out where some problem areas were. I've fixed the inconsistencies and made the confusing labels more clear.

### CODE REVIEW

**Comment by jhawthorn** [comment]
:+1: Looks good to me

File: core/config/locales/en.yml [hunk 1]
```diff
@@ -476,8 +476,8 @@ en:
         one: Tax Rate
         other: Tax Rates
       spree/taxon:
-        one: Taxon
-        other: Taxons
+        one: Display Order
+        other: Display Order
```
Comment (gmacdougall) [inline core/config/locales/en.yml]: I don't believe that this change is safe to make.

Comment (Mandily) [inline core/config/locales/en.yml]: Safe from a technical or user perspective? I'm not sure of an alternative if it's code, but if it's the user you're worried about, we could use an interim 'Display Order (Taxon)' label that would be intuitive for familiar users and new users alike.

**Comment by gmacdougall** [comment]
There are some failing tests now that the Taxon translation change has been reverted. Other than that, this is looking good.

**Pull request merged**